addons/token-exchange/token_exchanger_agent.go

```diff
@@ -156,9 +160,8 @@ func (o *AgentOptions) RunAgent(ctx context.Context, controllerContext *controll
 	go spokeKubeInformerFactory.Start(ctx.Done())
 	go greenSecretAgent.Run(ctx, 1)
 	go blueSecretAgent.Run(ctx, 1)
-	go leaseUpdater.Start(ctx)
 	runManager(ctx, hubRestConfig, controllerContext.KubeConfig, o.SpokeClusterName)
-
+	go leaseUpdater.Start(ctx)
```

Comment:
`runManager` is a blocking function. This line will not be trigerred until manager stops.

Reply:
Yeah this got mixed up when i was organizing the commits. Changed it